src/components/Tag/Tag.js

```diff
@@ -129,36 +122,69 @@ const TagElement = styled('span')`
 const Tag = ({
   children,
   icon,
+  renderPrefix: RenderPrefix,
+  renderSuffix: RenderSuffix,
   onRemove,
   labelRemoveButton,
   selected,
   ...props
-}) => (
-  <TagElement {...{ selected, ...props }}>
-    {!onRemove && icon && (
-      <IconContainer {...{ selected }}>{icon}</IconContainer>
-    )}
-    {children}
-    {onRemove && (
-      <CloseButton
-        onClick={onRemove}
-        selected={selected}
-        label={labelRemoveButton}
-        data-testid="tag-close"
-      />
-    )}
-  </TagElement>
-);
+}) => {
+  const prefix = RenderPrefix && <RenderPrefix />;
+  const suffix = RenderSuffix && <RenderSuffix />;
```

Comment:
The idea behind using a render prop for the icons was to pass the styles as a prop to the component to enable developers to override the styles if necessary. This isn't possible if the styles are applied to a container that wraps the prefix/suffix component.

Have another look at the [Input](https://github.com/sumup-oss/circuit-ui/blob/canary/src/components/Input/Input.js#L287) component and update to that approach, please :)

Reply:
Tbh, I disagree with the approach used in the `Input` component. The usual use case for the `renderPrefix` and `renderSuffix` props (especially in `Tag` or `Input`) would be to display some icon. The way we are passing `className` at the moment implies using boilerplate code every time one needs to add an icon somewhere which should be a rather simple task. The way I have implemented it now also allows overriding the styles of the prefix/suffix and makes passing a simple icon to the component much easier. 

Using the component this way:

`<Tag
    selected
    renderSuffix={({ className }) => (
      <span className={className}>
        <Icon />
      </span>
    )}
  >
    Transactions
  </Tag>`

seems way harder than: 

 `<Tag renderSuffix={Icon}>Transactions</Tag>`

or if you want to override the styles: 

 `<Tag renderSuffix={() => <StyledIcon /> }>Transactions</Tag>`

Reply:
> The way I have implemented it now also allows overriding the styles of the prefix/suffix and makes passing a simple icon to the component much easier.

It makes it possible to style the _icon_, but not the [_container_](https://github.com/sumup-oss/circuit-ui/pull/566/files#diff-a105d87b96967196f517996e7f609cfdR115) it's currently wrapped in. 

I agree that render props aren't ergonomic to use 😕 I like your solution that allows a simple icon to be passed but also offers the escape hatch to pass a function. I like it a lot. 

Would it be possible to use your approach _and_ get rid of the IconContainer? That'd be ideal in my opinion. In that case, I'd also simplify the prop names to `prefix` and `suffix` since they're no longer true render props.